pkg/internal/itest/mocknet/mocknet.go

```diff
@@ -0,0 +1,245 @@
+			} else {
+				qp.MapEntry(ma, "Status", qp.Int(int64(retrievalmarket.QueryResponseUnavailable)))
+				qp.MapEntry(ma, "PieceCIDFound", qp.Int(int64(retrievalmarket.QueryItemUnavailable)))
+				qp.MapEntry(ma, "Size", qp.Int(0))
+				qp.MapEntry(ma, "PaymentAddress", qp.Bytes([]byte{}))
+				qp.MapEntry(ma, "MinPricePerByte", qp.Bytes([]byte{}))
+				qp.MapEntry(ma, "MaxPaymentInterval", qp.Int(0))
+				qp.MapEntry(ma, "MaxPaymentIntervalIncrease", qp.Int(0))
+				qp.MapEntry(ma, "Message", qp.String(""))
+				qp.MapEntry(ma, "UnsealPrice", qp.Bytes([]byte{}))
+			}
+		})
+		require.NoError(t, err)
+		require.NoError(t, ipld.EncodeStreaming(s, queryResponse, dagcbor.Encode))
+		require.NoError(t, s.Close())
+	})
+}
+
+func (mrn *MockRetrievalNet) SetupRetrieval(ctx context.Context, t *testing.T) {
+	// Setup remote datastore and blockstore
+	dsRemote := dss.MutexWrap(datastore.NewMapDatastore())
+	dtDsRemote := namespace.Wrap(dsRemote, datastore.NewKey("datatransfer"))
+	bsRemote := bstore.NewBlockstore(namespace.Wrap(dsRemote, datastore.NewKey("blockstore")))
+	mrn.LinkSystemRemote = storeutil.LinkSystemForBlockstore(bsRemote)
+
+	// Setup remote data transfer
+	gsNetRemote := gsnet.NewFromLibp2pHost(mrn.HostRemote)
+	dtNetRemote := dtnet.NewFromLibp2pHost(mrn.HostRemote, dtnet.RetryParameters(0, 0, 0, 0))
+	gsRemote := gsimpl.New(ctx, gsNetRemote, mrn.LinkSystemRemote)
+	gstpRemote := gstransport.NewTransport(mrn.HostRemote.ID(), gsRemote)
+	dtRemote, err := dtimpl.NewDataTransfer(dtDsRemote, dtNetRemote, gstpRemote)
+	require.NoError(t, err)
+
+	// Wait for remote data transfer to be ready
+	dttestutil.StartAndWaitForReady(ctx, t, dtRemote)
+
+	// Register DealProposal voucher type with automatic Pull acceptance
+	remoteDealValidator := &mockDealValidator{acceptPull: true}
+	require.NoError(t, dtRemote.RegisterVoucherType(retrievalmarket.DealProposalType, remoteDealValidator))
+
+	// Record remote events
+	subscriberRemote := func(event datatransfer.Event, channelState datatransfer.ChannelState) {
+		mrn.RemoteEvents = append(mrn.RemoteEvents, event)
+		if event.Code == datatransfer.CleanupComplete {
+			mrn.FinishedChan <- struct{}{}
+		}
+	}
+	dtRemote.SubscribeToEvents(subscriberRemote)
+}
+
+func (mrn *MockRetrievalNet) WaitForFinish(ctx context.Context, t *testing.T) {
+	require.Eventually(t, ChanCheck(ctx, t, mrn.FinishedChan), 1*time.Second, 100*time.Millisecond)
+}
+
+func (mrn *MockRetrievalNet) TearDown() error {
+	return mrn.MN.Close()
+}
+
+var _ datatransfer.RequestValidator = (*mockDealValidator)(nil)
+
+type mockDealValidator struct {
+	acceptPull bool
+}
+
+func (mdv *mockDealValidator) ValidatePush(
+	channel datatransfer.ChannelID,
+	sender peer.ID,
+	voucher datamodel.Node,
+	baseCid cid.Cid,
+	selector datamodel.Node,
+) (datatransfer.ValidationResult, error) {
+	return datatransfer.ValidationResult{Accepted: false}, errors.New("not supported")
+}
+
+func (mdv *mockDealValidator) ValidatePull(
```

Comment:
maybe we need more complicated stuff over time? But love it's actually this simple (assumed it'd be a mess)